core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -320,6 +320,10 @@ void contractNodes() {
 
                 prepareGraph.disconnect(vehicleAllTmpExplorer, iter);
             }
+
+            if(Thread.currentThread().isInterrupted()){
```

Comment:
This is outside of the main loop, we would need it inside. Best would be directly after the while start to include the cases where `continue` is executed.